.gitattributes

```diff
@@ -0,0 +1,2 @@
+# Always use LF line endings even on Windows.
+* text eol=lf
```

Comment:
[Aside] - my understanding of the `eol=lf` part is that it'll force LF in everybody's working tree. I don't think we really care about this, although it's probably just fine to do so. But text files [should _always_ use LF endings in the index](https://git-scm.com/docs/gitattributes#_text). 

I do remember getting mixed line endings when checking out certain commits in the past, but now I can't imagine _how_ this could have happened. I wouldn't have had `config core.eol` set. Maybe the files weren't being classed as `text` by git? Have you ever had this happen / know why it does?

Reply:
>  I don't think we really care about this

Oh, this is answered in https://github.com/nextstrain/seasonal-flu/commit/202263aecf47e1906875bd2d2e80d7b0c82039cd

> Avoid situations like this one [1] where Windows users who clone this
repository would get TSV files with Windows line endings that then
caused tools in our workflow (like tsv-utils) to fail.

Reply:
> I don't think we really care about this

Ah yeah, I should have linked to the relevant threads on Slack: [mpox](https://bedfordlab.slack.com/archives/C03G8HQEV18/p1654791939251909?thread_ts=1654708786.585719&cid=C03G8HQEV18) and [seasonal-flu](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1711555334696739). 

> But text files [should always use LF endings in the index](https://git-scm.com/docs/gitattributes#_text).
>
> I do remember getting mixed line endings when checking out certain commits in the past, but now I can't imagine how this could have happened. I wouldn't have had config core.eol set. Maybe the files weren't being classed as text by git? Have you ever had this happen / know why it does?

My reading of the docs is text files always use LF endings **only** if the gitattribute is set. So, if the repo does not have `gitattributes` and a user does not have any local git configs, then they could have committed a file with CRLF endings. Then when you pulled down the file it retained the CRLF endings because you don't have `gitattributes` or local git configs to auto convert it to LF.

Reply:
Yeah, maybe you're right. The docs aren't clear when I read them again. Perhaps this is the most salient line:

> If the text attribute is unspecified, Git uses the core.autocrlf configuration variable to determine if the file should be converted.

The `autocrlf` [config](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration) is also rather complex, so I won't claim to understand all the various permutations.